test_programs/plonky2_prove_success/factorization/src/factor.nr

```diff
@@ -0,0 +1,89 @@
+// factor.nr
+pub fn print_factors(mut number: u32) {
+    // the number of primes limits the input to (59*59)^2 - 1 = 12117360
+    // the maximum factor limits the input to 256*256 - 1 = 65535
+    assert(number < 65536);
+    let mut factor_is_first = true;
+    //if number < 0 {
+    //    number = number * -1;
+    //    println(-1);
+    //    factor_is_first = false;
+    //}
+
+    // at most 16 primes
+    let mut primes: [u32; 0x10] = [0; 0x10];
+    let mut num_of_primes = 0;
+
+    // The first 16 primes are enough to figure out whether numbers up to 3480
+    // are prime; the 17th prime is 59, which squared is 3481.
+    for factor in 2..256 {
```

Comment:
Please, change 256 to 64; it reduces the runtime from 15 secs to 3 secs, which would be preferable since we run the tests before every PR (at least).

Reply:
Done: https://github.com/blocksense-network/noir/pull/56/commits/4c0c75ddd33eaae824e5bc9f1d8c359d9680dbad